dev/RadioButtons/ColumnMajorUniformToLargestGridLayout.cpp

```diff
@@ -138,6 +131,19 @@ void ColumnMajorUniformToLargestGridLayout::OnMaxColumnsPropertyChanged(const wi
     InvalidateMeasure();
 }
 
+int ColumnMajorUniformToLargestGridLayout::CalculateColumns(int childCount, float maxItemWidth, float availableWidth)
+{
+    auto const maxColumns = MaxColumns();
+    auto const columnSpacing = ColumnSpacing();
+
+    auto const extraSpaceAfterFirstColumn = availableWidth - maxItemWidth;
+    auto const maxExtraColumns = std::max(0.0, std::floor(extraSpaceAfterFirstColumn / (columnSpacing + maxItemWidth)));
```

Comment:
I'm a bit lost in this function. Could you add some comments for how the calculation it's doing? From the variable names alone I don't understand the approach.

Reply:
How's this look?

Reply:
Thanks, the new ascii art is great! :)